tests/common/system_utils/docker.py

```diff
@@ -0,0 +1,157 @@
+"""
+    docker contains utilities for interacting with Docker on the DUT.
+"""
+
+import collections
+import logging
+import os
+import time
+import yaml
+
+from common.broadcom_data import is_broadcom_device
+from common.mellanox_data import is_mellanox_device
+
+_LOGGER = logging.getLogger(__name__)
+
+_LOGIN_CMD = "docker login -u {} -p {} {}"
+_PULL_CMD = "docker pull {}/{}:{}"
+_RM_CMD = "docker rm {}"
+_TAG_CMD = "docker tag {}:{} {}"
+
+_BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+SONIC_DOCKER_REGISTRY = os.path.join(_BASE_DIR, "../../../ansible/vars/docker_registry.yml")
+
+_DockerRegistryInfo = collections.namedtuple('DockerRegistryInfo', 'host username password')
+class DockerRegistryInfo(_DockerRegistryInfo):
+    """
+        DockerRegistryInfo holds all the data needed to access a remote Docker registry.
+
+        Attributes:
+            host (str): The remote host where the Docker registry is located.
+            username (str): The username used to access the registry.
+            password (str): The password used to access the registry.
+    """
+    pass
+
+def parse_registry_file(registry_file):
+    """
+        parse_registry_file parses the provided file to produce a DockerRegistryInfo.
+
+        See `SONIC_DOCKER_REGISTRY` for the expected format of this file.
+
+        Args:
+            registry_file (str): The name of the file holding the registry information.
+
+        Raises:
+            IOError: If the file cannot be opened for any reason.
+            ValueError: If the provided file is missing any required fields.
+
+        Returns:
+            DockerRegistryInfo: The registry info from the registry file.
+    """
+
+    try:
+        with open(registry_file) as contents:
+            registry_vars = yaml.safe_load(contents)
+    except IOError as err:
+        _LOGGER.error("Failed to parse registry file \"%s\" (%s)", registry_file, err)
+        raise
+
+    host = registry_vars.get("docker_registry_host")
+    username = registry_vars.get("docker_registry_username")
+    password = registry_vars.get("docker_registry_password")
+
+    if not host or not username or not password:
+        error_message = "Registry file \"{}\" is missing login or hostname".format(registry_file)
+        _LOGGER.error(error_message)
+        raise ValueError(error_message)
+
+    return DockerRegistryInfo(host, username, password)
+
+def delete_container(dut, container_name):
+    """
+        delete_container attempts to delete the specified container from the DUT.
+
+        Args:
+            dut (SonicHost): The target device.
+            container_name (str): The name of the container to delete.
+    """
+
+    dut.command(_RM_CMD.format(container_name))
```

Comment:
here is a module for container from Ansible:

https://docs.ansible.com/ansible/latest/modules/docker_container_module.html#docker-container-module